src/support/archive.cpp

```diff
@@ -133,7 +133,7 @@ Archive::Child Archive::Child::getNext(bool& error) const {
   // Members are aligned to even byte boundaries.
   if (toSkip & 1) ++toSkip;
   const uint8_t* nextLoc = data + toSkip;
-  if (nextLoc >= (uint8_t*)&*parent->data.end()) {  // End of the archive.
+  if (nextLoc >= (const uint8_t*)parent->data.data() + parent->data.size()) {  // End of the archive.
```

Comment:
That kind of `if()` statement reads a bit cryptic. Pushed a bit simplified form that should be easy to read.